### PR TITLE
LANG-1717: support RISC-V in ArchUtils

### PR DESCRIPTION
**Pull request by kxxt**
Add RISC-V support to ArchUtils and its dependency class, Processor.

### CODE REVIEW

**Comment by garydgregory** [comment]
Hello @kxxt 
Do these strings match sources in https://github.com/openjdk/riscv-port ?

**Comment by kxxt** [comment]
> Do these strings match sources in https://github.com/openjdk/riscv-port ?

Hi @garydgregory, I get the `riscv64` value used in `addProcessors(new Processor(Processor.Arch.BIT_64, Processor.Type.RISCV), "riscv64")`  by evaluating `SystemProperties.getOsArch()` on openjdk17 installed on riscv64 linux. The `riscv32` counterpart is inferred.

As for the architecture name `RISCV` used in `Processor.Type`, I don't have any preference about the spelling and casing, `Riscv` is also okey.

**Comment by kxxt** [comment]
> Do these strings match sources in https://github.com/openjdk/riscv-port ?

Could you tell me where should I look for these strings?

It matches the values found here: https://github.com/openjdk/riscv-port/blob/64f8253b7d9d382b18003c647a3941733f191329/make/autoconf/platform.m4#L129-L139

File: src/test/java/org/apache/commons/lang3/ArchUtilsTest.java [hunk 1]
```diff
@@ -40,6 +40,8 @@ public class ArchUtilsTest extends AbstractLangTest {
     private static final String X86 = "x86";
     private static final String X86_64 = "x86_64";
     private static final String AARCH_64 = "aarch64";
+    private static final String RISCV64 = "riscv64";
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/lang3/ArchUtilsTest.java:43]: @kxxt 
Thank you for the update.
Follow the style in this class: "RISCNUM" -> "RISC_NUM".

Comment (kxxt) [reply src/test/java/org/apache/commons/lang3/ArchUtilsTest.java]: @garydgregory  Thanks for your quick review. It is fixed now.

**Comment by kxxt** [comment]
I have resolved the merge conflict.

And by the way, if it matters, Apache has received my ICLA:

> This message acknowledges receipt of your ICLA, which has been filed in the Apache Software Foundation records.
> 
> With this message, the Commons PMC has been notified that your ICLA has been filed.

**Comment by garydgregory** [comment]
Hi @kxxt 
Thank you for your contribution.

**Pull request merged**